### PR TITLE
Abort sendbuffer loop if the TCP socket closes midway

### PR DESCRIPTION
**Pull request by SwadicalRag**

### CODE REVIEW

**Comment by SwadicalRag** [comment]
why is this branch so behind

**Comment by SwadicalRag** [comment]
wait what

**Comment by danielga** [comment]
@SwadicalRag you need to rebase on master.

File: lua/websocket/server.lua [hunk 1]
```diff
@@ -257,6 +257,14 @@ function SERVER:Think()
 				if connection.errorcallback ~= nil then
 					connection.errorcallback(connection, "websocket auth error: " .. err)
 				end
+				
+				if err == "closed" then
+					connection.state = CLOSED
+					if connection.errorcallback ~= nil then
+						connection.closecallback(connection, 1006, "socket closed")
```
Comment (danielga) [inline lua/websocket/server.lua:264]: You use `closecallback` but check `errorcallback` for nil.

**Review by danielga — CHANGES_REQUESTED** [review]

**Review by danielga — APPROVED** [review]

**Pull request merged**